plugins/modules/ipbase_info.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, Dominik Kukacka <dominik.kukacka@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: ipbase_info
+version_added: 7.0.0
+short_description: Retrieve IP geolocation and other facts of a host's IP address
+description:
+  - "Retrieve IP geolocation and other facts of a host's IP address using the ipbase.com API"
+author: "Dominik Kukacka (@dominikkukacka)"
+extends_documentation_fragment:
+  - community.general.attributes
+  - community.general.attributes.info_module
+options:
+  ip:
+    description:
+      - "The IP you want to get the info for."
+    required: false
+    type: str
+  apikey:
+    description:
+      - "The apikey for the request if you need more requests."
+    required: false
+    type: str
+notes:
+  - "Check U(https://ipbase.com/) for more information."
+'''
+
+EXAMPLES = '''
+- name: "Get IP geolocation information of the primary outgoing IP"
+  community.general.ipbase_info:
+  register: my_ip_info
+
+- name: "Get IP geolocation information of a specific IP"
+  community.general.ipbase_info:
+    ip: "8.8.8.8"
+  register: my_ip_info
+'''
+
+RETURN = '''
+data:
+    description: "The data retrieved from ipbase.com."
+    returned: success
+    type: dict
```

Comment:
An earlier version of the PR was documenting all the fields. Could you please put that information back?
```suggestion
    type: dict
    contains:
       ...
```

Reply:
I moved to mirror the API response of ipbase.com.
So I mentioned now that the `data` attribute is only returned on success and refers the user to the documentation for more information about the response.